mmdet/datasets/custom.py

```diff
@@ -322,3 +322,27 @@ def evaluate(self,
                 for i, num in enumerate(proposal_nums):
                     eval_results[f'AR@{num}'] = ar[i]
         return eval_results
+
+    def __repr__(self):
+        """Print the number of instance number."""
+
+        if self.CLASSES is None:
+            return "Dataset is empty"
+        instance_count = np.zeros(len(self.CLASSES)).astype(int)
+        # count the instance number in each image
+        for idx in range(len(self)):
+            unique, counts = np.unique(self.get_cat_ids(idx),
+                                       return_counts=True)
+            if len(unique) > 0:
+                instance_count[unique] += counts
```

Comment:
This writing is incorrect because the `cat_ids` of the coco dataset may be larger than `self.CLASSES`

Reply:
I do not take the `cat_ids` into consideration other than the classes in `self.CLASSES`. I would only want to calculate the instance numbers within `self.CLASSES`.